main.py

```diff
@@ -88,6 +89,17 @@ def knapsack_bqm(costs, weights, weight_capacity):
 
 sampler = LeapHybridSampler()
 sampleset = sampler.sample(bqm)
-solution = [df['weight'][i] for i in range(len(df['weight'])) if sampleset.record.sample[0][i] == 1.]
-
-print("Found solution {} at energy {}.".format(solution, sampleset.first.energy))
+n_sols = sampleset.record.sample.shape[0]
+for i in range(n_sols):
+
+    # Build solution from returned bitstring
+    solution = []
+    for j0, j in enumerate(sampleset.record.sample[i]):
```

Comment:
perhaps use a more expressive variable naming scheme? perhaps `j` for the indexing and `sample` for the sample?

This way, it's easier for someone who is simply reading the code to understand what is happening

Reply:
Thanks @m3ller. I changed them. It's possible there will be a better way to write it. I'm a little leery of some tricks because the HSS returns floating-point 1's, so I am trying to be careful.